### DISCUSSION TITLE
full text search engine : just for learning

### DISCUSSION THREAD

**Discussion by ats1999** [scaleracademy/open-source-projects#89] [Ideas]
## What is full-text search engine
> Refer to the below docs

- https://en.wikipedia.org/wiki/Full-text_search
- https://www.techopedia.com/definition/17113/full-text-search
- https://www.couchbase.com/products/full-text-search

## Why I want to build this
While building a full-text search engine, we'll get to implement real-life algorithms, scaling concepts, etc.

## What I want
If anyone is interested in building this  project along with me, then comment on this discussion

**Comment by ats1999** [comment]
@championswimmer please have a look

**Reply by rahul-works** [reply]
which tech stack?

**Reply by ats1999** [reply]
It, depends on the people who will be working and the best suited tech  stack would be.

**Reply by prakhsrivastava** [reply]
I am in, wanna see how we are going to implement this.

**Reply by ats1999** [reply]
@prakhsrivastava welcome

**Reply by chandanjon** [reply]
i am in

**Reply by ats1999** [reply]
@chandanjon @prakhsrivastava may i know what tech stack do you guys know, if possible then rate yourself in that tech stack on a scale of 5.

## Me
- NodeJS 9/10
- ReactJs 9/10
- Java 5/10 - I know only programming, not any framework based on java

**Reply by prakhsrivastava** [reply]
@ats1999 

<h2>Here's Mine</h2>
<ul>
  <li>PHP 4/5</li>
  <li>Laravel 4/5</li>
  <li>jQuery 4/5</li>
  <li>ReactJs 1/5 - I know some basics</li>
</ul>

**Reply by ats1999** [reply]
@prakhsrivastava i think, it's not good to develop in php, it would be better to develop in java, node or python

**Reply by rajatpal96** [reply]
Hi i am in , i wanna do this

**Comment by prakhsrivastava** [comment]
That i can understand Rahul but as a platform it should work properly or
just remove it from there.

**Reply by ats1999** [reply]
As a platform, it'll work properly. It'll be deployed and open source. Also, I've mentioned that `full text search engine : just for learning`

**Comment by ats1999** [comment]
Upvote it if you would like to work on this.

**Comment by prakhsrivastava** [comment]
Sorry rahul i thought the message was from scaler helpdesk because there
was a problem with PHP.

**Reply by ats1999** [reply]
ohh, no issues

**Comment by prakhsrivastava** [comment]
Java will be fine i think.

**Comment by rkunnattur** [comment]
I am interested in this Idea
Have experience in Java and Java-related frameworks basically, but am open to picking up and working on other tech stacks like  Javascript or Python frameworks too

**Reply by ats1999** [reply]
Welcome @rkunnattur 

yeah, can we pick NodeJs?

**Comment by ats1999** [comment]
@rkunnattur @prakhsrivastava @rahul-works @chandanjon can we start

**Reply by rkunnattur** [reply]
Yes

**Comment by prakhsrivastava** [comment]
Yes we can.

**Reply by ats1999** [reply]
Let's wait for few more people

**Comment by tarunngusain08** [comment]
Python - 4/5
Java - 3/5
React - 2/5
If you are opting Node.js. I would be still interested in the project😊

**Reply by ats1999** [reply]
It's good to know that you are interested, but I am looking for some exp developer to guide us. 

@championswimmer  can you??

**Comment by rkunnattur** [comment]
Started?

**Reply by ats1999** [reply]
https://github.com/ats1999/full-text-search-engine/issues/1 

Comment on the issue

**Comment by ats1999** [comment]
Those who are interested can comment on the issue https://github.com/ats1999/full-text-search-engine/issues/1

**Comment by dilipdmane** [comment]
I am interested and here are my skill set

let me know 

java: 7/10
Java script: 5/10
NodeJs : 3/10
SQL:5/10
Mongodb :3/10
Spring framework: 4/10
